editor/components/block-select-parent/index.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { omit } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/element';
+import { ifCondition, IconButton, Tooltip, withSafeTimeout } from '@wordpress/components';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+export function BlockSelectParent( { selectRootBlock, onFocus, ...props } ) {
+	const selectParentLabel = __( 'Select parent block' );
+	const onFocusHandler = ( event ) => {
+		if ( onFocus ) {
+			onFocus( event );
+		}
+		// This is used for improved interoperability
+		// with the block's `onFocus` handler which selects the block, thus conflicting
+		// with the intention to select the root block.
+		event.stopPropagation();
+	};
+
+	return (
+		<Tooltip text={ selectParentLabel }>
+			<IconButton
+				onClick={ selectRootBlock }
+				onFocus={ onFocusHandler }
+				label={ selectParentLabel }
+				icon="arrow-left-alt"
+				{ ...omit( props, [ 'rootUID', 'selectRootBlock', 'setTimeout', 'clearTimeout' ] ) }
+			/>
+		</Tooltip>
+	);
+}
+
+export default compose( [
+	withSelect( ( select, ownProps ) => {
+		const { getBlockRootUID } = select( 'core/editor' );
+		const { uid } = ownProps;
+
+		return {
+			rootUID: getBlockRootUID( uid ),
+		};
+	} ),
+	ifCondition( ( { rootUID } ) => rootUID ),
+	withSafeTimeout,
+	withDispatch( ( dispatch, ownProps ) => {
+		const { rootUID, setTimeout } = ownProps;
+		const { selectBlock } = dispatch( 'core/editor' );
+
+		return {
+			// execute after all event handlers.
+			selectRootBlock: () => setTimeout( () => selectBlock( rootUID ), 0 ),
```

Comment:
This sounds a little bit hacky, I tried many other options, but in all of them, a strange bug happened where after the select root action was dispatched a new action was dispatched to select the first child of the root. Making the first child in the nested container the selected one. Using the timeOut things works great.

Reply:
I believe there are various occurrences of `setTimeout()` used this way in the codebase, even omitting the delay value, which I guess has the same effect of "moving execution to the end of the queue".